File: thirdSeminar/src/main/java/org/sopt/thirdSeminar/domain/CategoryId.java
package org.sopt.thirdSeminar.domain;

import jakarta.persistence.Embeddable;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Embeddable
@Getter
@EqualsAndHashCode
@NoArgsConstructor
public class CategoryId implements Serializable {

    private String categoryId;

    @Builder
    private CategoryId(String categoryId) {
        this.categoryId = categoryId;
    }
}

File: sixthKotlin/src/main/kotlin/org/sopt/sixthKotlin/domain/CategoryId.kt
package org.sopt.sixthKotlin.domain

import jakarta.persistence.Embeddable
import lombok.Builder
import lombok.EqualsAndHashCode
import java.io.Serializable

@Embeddable
@EqualsAndHashCode
class CategoryId @Builder private constructor(val categoryId: String) : Serializable